flag.go

```diff
@@ -14,26 +14,30 @@ const (
 )
 
 type FeatureFlag struct {
-	Key           string         `json:"key" bson:"key"`
-	Version       int            `json:"version" bson:"version"`
-	On            bool           `json:"on" bson:"on"`
-	Prerequisites []Prerequisite `json:"prerequisites" bson:"prerequisites"`
-	Salt          string         `json:"salt" bson:"salt"`
-	Sel           string         `json:"sel" bson:"sel"`
-	Targets       []Target       `json:"targets" bson:"targets"`
-	Rules         []Rule         `json:"rules" bson:"rules"`
-	Fallthrough   Rule           `json:"fallthrough" bson:"fallthrough"`
-	OffVariation  *int           `json:"offVariation" bson:"offVariation"`
-	Variations    []interface{}  `json:"variations" bson:"variations"`
-	Deleted       bool           `json:"deleted" bson:"deleted"`
+	Key           string             `json:"key" bson:"key"`
+	Version       int                `json:"version" bson:"version"`
+	On            bool               `json:"on" bson:"on"`
+	Prerequisites []Prerequisite     `json:"prerequisites" bson:"prerequisites"`
+	Salt          string             `json:"salt" bson:"salt"`
+	Sel           string             `json:"sel" bson:"sel"`
+	Targets       []Target           `json:"targets" bson:"targets"`
+	Rules         []Rule             `json:"rules" bson:"rules"`
+	Fallthrough   VariationOrRollout `json:"fallthrough" bson:"fallthrough"`
+	OffVariation  *int               `json:"offVariation" bson:"offVariation"`
+	Variations    []interface{}      `json:"variations" bson:"variations"`
+	Deleted       bool               `json:"deleted" bson:"deleted"`
 }
 
-// Expresses a set of AND-ed matching conditions for a user, along with
-// either the fixed variation or percent rollout to serve if the conditions
-// match.
-// Invariant: one of the variation or rollout must be non-nil.
+// Expresses a set of AND-ed matching conditions for a user, along with either a fixed
+// variation or a set of rollout percentages
 type Rule struct {
-	Clauses   []Clause `json:"clauses" bson:"clauses"`
+	VariationOrRollout
```

Comment:
I believe you will need `bson:",inline"`, otherwise this will change the document layout in mongo.